src/combinatorics/simple_graph/subgraph.lean

```diff
@@ -61,9 +61,27 @@ structure subgraph {V : Type u} (G : simple_graph V) :=
 (edge_vert : ∀ {v w : V}, adj v w → v ∈ verts)
 (symm : symmetric adj . obviously)
 
+variables {V : Type u} {W : Type v}
+
+/-- The one-vertex subgraph. -/
+@[simps]
+protected def singleton_subgraph (G : simple_graph V) (v : V) : G.subgraph :=
```

Comment:
Have you considered making this
```suggestion
instance : has_singleton G.subgraph :=
```
To me this is motivated by the fact that the subgraphs of that form are precisely atoms of the subgraph lattice. This will however lead to more type-ascriptions.

Reply:
I get where you're coming from here, but it seems like re-using singleton notation isn't quite right since subgraphs aren't sets per se (though they're also sets with higher structure). I worry also that using singleton notation would make learning to use the graph API more of a challenge than it already is.

Just so you know what the idea is with this PR: when you have these two types of subgraphs you can easily turn a walk into a subgraph using sup, and then we can define whether a subgraph is a path or a cycle depending on whether it equals a subgraph associated to a `walk.path` or a `walk.cycle`.

Reply:
Yeah, this is very much an optional suggestion. I agree it will probably be impractical.